ONBOARD.md

```diff
@@ -0,0 +1,183 @@
+a `tests` array (so it is possible to specify multiple test targets). Here is an
+example of two test targets, each performing a different test by calling
+different `make` target in a `src` image (of course, a `Makefile` in your
+component repository would need to have these targets for this to work).
+
+```json
+{
+  "tests": [
+    {
+      "as": "unit",
+      "from": "src",
+      "commands": "make test-unit"
+    },
+    {
+      "as": "performance",
+      "from": "src",
+      "commands": "make test-performance"
+    }
+  ]
+}
+```
+
+By default, ci-operator runs all specified test targets, building all their
+dependencies (and transitively, their dependencies) before. You can limit the
+execution to build just one or more targets using the `--target` option.
+
+### Intermediary binary targets
+
+Two test targets in the previous example assume their `make` targets take care
+of full build from source till the actual test. This is often the case, but it
+is ineffient because each test target performs the build separately. CI
+operator can create `bin` and `test-bin` targets for the test targets to share
+by providing `binary_build_commands` and `test_binary_build_commands`
+respectively (we have two test lists here because often it's a different
+compilation process for test binaries than for normal ones -- in Go that is the
+difference between "normal" compilation and compilation to test race
+conditions):
+
+```json
+{
+  “binary_build_commands”: “make build”,
+  “test_binary_builds_commands”: “make instrumented-build”,
+  “tests”: [
+    {
+      “as”: “unit”,
+      “from”: “bin”,
+      “commands”: “make test-unit”
+    },
+    {
+      “as”: “integration”,
+      “from”: “bin”,
+      “commands”: “make test-integration”,
+    },
+    {
+      “as”: “performance”,
+      “from”: “test-bin”,
+      “commands”: “make test-performance”
+    }
+  ]
+}
+```
+
+Here, `unit` and `integration` targets will both be built from a `bin` image,
+which will be a result of running `make build` over a `src` image.
+
+### Submit the configuration file to openshift/release
+
+When you describe the targets for your component in the configuration file, you
+will need to add the file to the
+[openshift/release](https://github.com/openshift/release) repository,
+specifically to its `ci-operator/config/openshift` subdirectory
+[tree](https://github.com/openshift/release/tree/master/ci-operator/config/openshift).
+Each OpenShift component has a separate directory there, and there is a
+configuration file in it per branch.
+
+### Images targets, end-to-end tests and more
+
+Building the source code and running unit tests is the trivial basic use case.
+ci-operator is able to build component images, provision test clusters using
+them and run end-to-end tests on them. These use cases would use more features
+in both configuration file and Prow job and would not fit into this document.
```

Comment:
Just note that it will fit in later, as Bruno will likely edit this doc after you are done

Reply:
(also so that someone coming here to look for that information has some knowledge that it will be here later, rather than just knowing they've come to the wrong place without a pointer as to where else to look)